Utilities/Thread.cpp

```diff
@@ -2019,20 +2019,95 @@ u16 thread_ctrl::get_affinity_mask(thread_class group)
 		}
 		case native_core_arrangement::amd_ccx:
 		{
-			u16 spu_mask, ppu_mask, rsx_mask;
-			if (thread_count >= 16)
+			u32 spu_mask, ppu_mask, rsx_mask;
+			const auto system_id = utils::get_system_info();
+			if (thread_count >= 32)
 			{
-				// Threadripper, R7
-				// Assign threads 8-16
-				// It appears some windows code is bound to lower core addresses, binding 8-16 is alot faster than 0-7
-				ppu_mask = spu_mask = 0b1111111100000000;
-				rsx_mask = all_cores_mask;
+				if (system_id.find("3950X") != std::string::npos)
+				{
+					// zen2
+					// Ryzen 9 3950X
+					// Assign threads 9-30
+					ppu_mask = 0b1111000000000000000000000000;
+					spu_mask = 0b111111110000000000000000;
```

Comment:
```suggestion
					spu_mask = 0b0000111111110000000000000000;
```
pad the rest of the values like so, to improve readability

Reply:
hmm yes i know, i jave set all like that for can test, but i have after remove 0not needed for get like set on master with @kd-11